pandas/core/internals.py

```diff
@@ -4161,8 +4161,9 @@ def construction_error(tot_items, block_shape, axes, e=None):
         raise e
     if block_shape[0] == 0:
         raise ValueError("Empty data passed with indices specified.")
-    raise ValueError("Shape of passed values is {0}, indices imply {1}".format(
-        passed, implied))
+    raise ValueError("Shape of passed values is {0}, indices imply {1}.\n\
+        This means that at least one of the dataframes contains duplicate \
```

Comment:
this is not true at all, this error could be called for quite a few reasons. pls put the error more specifially (e.g. in merge.py) and test if there are duplicates.